mm2src/mm2_bitcoin/spv_validation/src/conf.rs

```diff
@@ -0,0 +1,109 @@
+}
+
+fn from_bit_u32<'de, D>(deserializer: D) -> Result<BlockHeaderBits, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let bits: u32 = Deserialize::deserialize(deserializer)?;
+    Ok(BlockHeaderBits::Compact(bits.into()))
+}
+
+/// Custom SPV starting block header configuration
+#[derive(Debug, Clone, Deserialize)]
+pub struct SPVBlockHeader {
+    /// Valid `u32` representation of the block `height`.
+    pub height: u64,
+    /// Valid block header `hash`.
+    #[serde(deserialize_with = "from_hash_str")]
+    pub hash: H256,
+    /// Valid `u32` representation of the `date` the block is mined in epoch.
+    pub time: u32,
+    /// Valid block header `bits`.
+    #[serde(deserialize_with = "from_bit_u32")]
+    pub bits: BlockHeaderBits,
+}
+
+impl SPVBlockHeader {
+    pub(crate) fn from_block_header(height: u64, header: BlockHeader) -> Self {
+        Self {
+            height,
+            hash: header.hash(),
+            time: header.time,
+            bits: header.bits,
+        }
+    }
+}
+
+/// Validate that `max_stored_headers_value` is always greater than `retarget interval`.
+fn validate_btc_max_stored_headers_value(max_stored_block_headers: u64) -> Result<(), SPVError> {
+    if RETARGETING_INTERVAL > max_stored_block_headers as u32 {
+        return Err(SPVError::SPVBlockHeaderError(format!(
+            "max_stored_block_headers {max_stored_block_headers} must be greater than retargeting interval {RETARGETING_INTERVAL}",
+        )));
+    }
+
+    Ok(())
+}
+
+/// Validate that starting block header is a retarget header.
+fn validate_btc_spv_header_height(coin: &str, height: u64) -> Result<(), SPVError> {
+    let is_retarget = height % RETARGETING_INTERVAL as u64;
+    if is_retarget != 0 {
+        return Err(SPVError::WrongRetargetHeight {
+            coin: coin.to_string(),
+            expected_height: height - is_retarget,
+        });
+    }
+
+    Ok(())
+}
+
+/// SPV headers verification parameters
+#[derive(Clone, Debug, Deserialize)]
+pub struct BlockHeaderValidationParams {
+    pub difficulty_check: bool,
+    pub constant_difficulty: bool,
+    pub difficulty_algorithm: Option<DifficultyAlgorithm>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct SPVConf {
+    /// Where to start block headers sync from.
+    pub block_header: SPVBlockHeader,
+    /// Max number of block headers to be stored in db, when reached, old header will be deleted.
+    pub max_stored_block_headers: Option<NonZeroU64>,
+    /// The parameters that specify how the coin block headers should be validated. If None,
+    /// headers will be saved in DB without validation, can be none if the coin's RPC server is trusted.
+    pub validation_params: Option<BlockHeaderValidationParams>,
+}
+
+impl SPVConf {
+    pub fn validate(&self, coin: &str) -> Result<(), SPVError> {
+        if let Some(params) = &self.validation_params {
+            if let Some(DifficultyAlgorithm::BitcoinMainnet) = &params.difficulty_algorithm {
```

Comment:
https://github.com/KomodoPlatform/atomicDEX-API/blob/1e9b0e13a8dd952693515dfdddf19aa714f01097/mm2src/mm2_bitcoin/spv_validation/src/conf.rs#L107
It's better to use `todo!()` or `unimplemented!()`, this will lead to panicking in the future if you forgot to implement this validation. Right now, this validation will just pass if you forgot to implement it when working on `DifficultyAlgorithm::BitcoinTestnet`.